Makefile

```
@@ -48,31 +46,24 @@ clean:
 
 .PHONY: list list-yaml
 list:
-	$(info $(PKGNAMELIST))
+	$(info $(PKGLIST))
 	@printf ''
 
 list-yaml:
-	$(info $(addsuffix .yaml,$(PKGNAMELIST)))
+	$(info $(addsuffix .yaml,$(PKGLIST)))
 	@printf ''
 
-.packagerules: Makefile .git/HEAD packages.txt
-	@echo "Solving build order, please wait..."
-	@grep -v '^\#' packages.txt | while read pkg; do		\
-		pkgname=`echo $$pkg | cut -d, -f1`;			\
-		pkgdir=`echo $$pkg | cut -d, -f2`;			\
-		[ -z "$$pkgdir" ] && pkgdir=$$pkgname;			\
-		pkgver=`${MELANGE} package-version $${pkgname}.yaml`;	\
-		pkgtarget="${TARGETDIR}/$${pkgver}.apk";		\
-		echo "PKGNAMELIST += $$pkgname";			\
-		echo ".build-packages: $$pkgtarget";			\
-		echo "packages/$$pkgname: $$pkgtarget";			\
-		echo "$$pkgtarget: $${pkgname}.yaml \$${KEY}";		\
-		printf "\t%s\n" "@mkdir -p ./$${pkgdir}/";		\
-		printf "\t%s" "SDE=\$${SOURCE_DATE_EPOCH}; [ -z \"\$$\$$SDE\" ] && SDE=\`git log -1 --pretty=%ct --follow $${pkgname}.yaml\`;"; \
-		printf "\t%s\n\n" "SOURCE_DATE_EPOCH=\$$\$$SDE \$${MELANGE} build $${pkgname}.yaml \$${MELANGE_OPTS} --source-dir ./$${pkgdir}/ --log-policy builtin:stderr,\$${TARGETDIR}/buildlogs/$${pkgver}.log"; \
-	done > .packagerules
+.build-packages: $(addprefix package/,${PKGLIST})
 
--include .packagerules
+package/%:
+	$(eval yamlfile := $*.yaml)
+	$(eval pkgver := $(shell $(MELANGE) package-version $(yamlfile)))
+	$(MAKE) yamlfile=$(yamlfile) pkgname=$* packages/$(ARCH)/$(pkgver).apk
+
+packages/$(ARCH)/%.apk: $(KEY)
+	@mkdir -p ./$(pkgname)/
+	$(eval SOURCE_DATE_EPOCH ?= $(shell git log -1 --pretty=%ct --follow $(yamlfile)))
+	@SOURCE_DATE_EPOCH=$(SOURCE_DATE_EPOCH) $(MELANGE) build $(yamlfile) $(MELANGE_OPTS) --source-dir ./$(pkgname)/ --log-policy builtin:stderr,$(TARGETDIR)/buildlogs/$*.log"
 
 dev-container:
 	docker run --privileged --rm -it -v "${PWD}:${PWD}" -w "${PWD}" ghcr.io/wolfi-dev/sdk:latest@sha256:3ef78225a85ab45f46faac66603c9da2877489deb643174ba1e42d8cbf0e0644
```

happy to do this in a different PR if you want, but somehow this `dev-container` got out of date enough to not include `wolfictl`, which, with this PR, is now needed for the dev loop

```bash
❯ crane export ghcr.io/wolfi-dev/sdk:latest@sha256:3ef78225a85ab45f46faac66603c9da2877489deb643174ba1e42d8cbf0e0644 | tar tvf - | grep -i wolfictl

```

vs:

```bash
❯ crane export ghcr.io/wolfi-dev/sdk:latest@sha256:3ba6e392eff7f09493c62b8a6bff4b9378ecccc27e5dc4ba0fa9f2a0e95c666f | tar tvf - | grep -i wolfictl
-rwxr-xr-x  0 root   root 72379332 Dec 31  1969 usr/bin/wolfictl
                                                                                                                                                               
```

```suggestion
	docker run --privileged --rm -it -v "${PWD}:${PWD}" -w "${PWD}" ghcr.io/wolfi-dev/sdk:latest@sha256:3ba6e392eff7f09493c62b8a6bff4b9378ecccc27e5dc4ba0fa9f2a0e95c666f
```

---

It probably is because we have digestabot update these files directly. If we had a single file with all of them, and the Makefile and workflows just pulled them in, it all would get updated automatically. Maybe I will open a PR for it.

In the meantime, I will rebase and update it.